### DISCUSSION TITLE
Override how Guid.ToString for Sqlite is implemented in SQL

### DISCUSSION THREAD

**Discussion by jogibear9988** [linq2db/linq2db#4296] [Q&A]
Is there a way, so I could override, how ToString() of a guid will be converted to SQL?

In Sqlite I need it to generate SQL like this:

        substr(hex(id), 1, 8)
        || '-' || substr(hex(id), 9, 4)
        || '-' || substr(hex(id), 13, 4)
        || '-' || substr(hex(id), 17, 4)
        || '-' || substr(hex(id), 21, 12)

**Comment by jogibear9988** [comment]
I could do it like this in SQLLiteQueryOptimizer, but is this the correct location?
And also I could not provide SQLiteDataProvider my own "SqliteOptimizer", without implementing also my own SQLiteDataProvider

<img width="1342" alt="image" src="https://github.com/linq2db/linq2db/assets/364896/2d28c735-1eb6-4593-9c28-7d6d7087e2f7">

**Comment by jogibear9988** [comment]
And also the SQLiteSqlOptimizer is sealed, so I need to copy the whole code

**Comment by MaceWindu** [comment]
literals generation logic stored in mapping schema. You need to register your generation logic using `SetValueToSqlConverter` method like [here](https://github.com/linq2db/linq2db/blob/master/Source/LinqToDB/DataProvider/SQLite/SQLiteMappingSchema.cs#L25)

**Comment by jogibear9988** [comment]
@MaceWindu but is this not the other direction, Guid to String? I need a string comparison of the stored guid in SQL.

I will create a pull with my change, if its okay

**Comment by sdanyliv** [comment]
I do not understand this logic. Why it is needed? Create your own extension method.

**Comment by jogibear9988** [comment]
because in SQLSever for example a guid could be converted to a string, in sqlite it could not
